pkg/controller/device/device_controller.go

```diff
@@ -89,5 +93,58 @@ func (r *ReconcileDevice) Reconcile(request reconcile.Request) (reconcile.Result
 		return reconcile.Result{}, err
 	}
 
+	err = r.CheckBackingDiskStatusAndUpdateDeviceCR(instance,
+		request.NamespacedName.Namespace, reqLogger)
+
+	if err != nil {
+		// Error while reading, updating object - requeue the request.
+		return reconcile.Result{}, err
+	}
 	return reconcile.Result{}, nil
 }
+
+func (r *ReconcileDevice) CheckBackingDiskStatusAndUpdateDeviceCR(
```

Comment:
Can the NDM daemon set take care of managing this relationship.

Reply:
@kmova If daemon set needs to manage the relationship, it needs to continuously monitor disk/devices. But currently NDM Daemon is triggered only by udev events.

Reply:
NDM Daemonset would be responsible to push disk and device CR in to etcd. Rest is taken care by device controller and disk controller. Each Device -CR knows identity of backing disk(if device-CR is backed by physical disk).